enumeratum-core/src/main/scala/enumeratum/EnumEntry.scala

```diff
@@ -61,4 +61,17 @@ object EnumEntry {
     abstract override def entryName: String = super.entryName.toLowerCase
   }
 
+  /**
+   * Helper implicit for more readable checking.
+   */
+  implicit class ComparableEnum[A <: EnumEntry](enum: A) {
```

Comment:
Ah, maybe I should have been more clear. I wasn't talking about the type constraint; moreso the implicit class.

By having the implicit class `extend AnyVal`, we can prevent a runtime allocation because the compiler will rewrite the code into a method call on a static object.